File: 250730/정수 N개의 합 2/sum-of-n-integers-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, k] = input[0].trim().split(' ').map(Number);
const arr = input[1].trim().split(' ').map(Number);

const temp = Array(n+1).fill(0);
temp[0] = arr[0]
for(let i = 1; i < n+1; i++){
    temp[i] = temp[i-1] + arr[i];
}

let max = Number.MIN_SAFE_INTEGER;
for(let i = k; i < n; i++){
    max = Math.max(max, temp[i] - temp[i-k]);
}

console.log(max)

File: 260131/정수 N개의 합 2/sum-of-n-integers-2.py
import sys
n, k = map(int, input().split())
arr = list(map(int, input().split()))

total_sum = [0] * (n + 1)
ans = -sys.maxsize

for i in range(1, n + 1):
    total_sum[i] = total_sum[i-1] + arr[i - 1]

for i in range(0, n - k + 1):
    ans = max(ans, total_sum[i + k] - total_sum[i])

print(ans)
